### ISSUE TITLE
Segfault when trying to open a previous instance with an LV2 plugin.

### ISSUE THREAD

**Issue by M0JXD** [Stazed/non-mixer-xt#48]
Hi Stazed,

This is the mixer's folder from a RaySession of mine. When trying to open it there is a crash at start up. The plugin it was hosting was x42's Red Zeppelin Drums (stereo).

[Non-Mixer-XT.m_xt.zip](https://github.com/user-attachments/files/18260659/Non-Mixer-XT.m_xt.zip)

Prior to using mixer for this I was using Carla. Funnily enough when I tried to use RaySession's git restore, Carla crashes at start with it's restored session files as well. So it's probably not be a Mixer problem. I've checked my update history and I can't see any updates for the plugin? I know LV2 is notorious for this...

Thought you should know regardless.

**Comment by Stazed** [comment]
The snapshot indicates that a custom data directory should be present :

> Red_Zepplin_Drumkit.nXQPM

The directory does not exist in the zip file you sent. Did you somehow delete it?

Regardless, the mixer should not crash when this error occurs so I will fix that.

**Comment by M0JXD** [comment]
I don't think so, but during Non-Sequencer testing I have been adding/removing unneeded clients (via RaySession's feature for it). I've found a git snapshot that works now but I'll try and find why it was deleted.

**Comment by Stazed** [comment]
Upon further investigation, even with the directory deleted, is does not crash for me as there is already error checking implemented. Can to supply a core dump? Does the mixer project load outside of raysession? Just open an instance of the mixer outside of raysession from terminal and select the mixer project folder from the session. When running from terminal, watch for any warnings, error messages.

**Comment by M0JXD** [comment]
If I choose that folder in a standalone instance outside of RaySession it still crashes.

Here's the output:

jamiedrinkell@jamiedrinkell-Nitro-AN515-57:~/Desktop$ ulimit -c unlimited
jamiedrinkell@jamiedrinkell-Nitro-AN515-57:~/Desktop$ non-mixer-xt 
The Non-Mixer-XT 2.0.6
Copyright (C) 2008-2021 Jonathan Moore Liles (as Non-Mixer)
Copyright (C) 2021- Stazed (as Non-Mixer-XT)
Cannot connect to server socket err = Connection refused
Cannot connect to server request channel
jackdmp 1.9.21
Copyright 2001-2005 Paul Davis and others.
Copyright 2004-2016 Grame.
Copyright 2016-2022 Filipe Coelho.
jackdmp comes with ABSOLUTELY NO WARRANTY
This is free software, and you are welcome to redistribute it
under certain conditions; see the file COPYING for details
no message buffer overruns
no message buffer overruns
no message buffer overruns
JACK server starting in realtime mode with priority 10
self-connect-mode is "Don't restrict self connect requests"
audio_reservation_init
Acquire audio card Audio1
creating alsa driver ... hw:PCH,0|hw:PCH,0|1024|2|48000|0|0|nomon|swmeter|-|32bit
configuring for 48000Hz, period = 1024 frames (21.3 ms), buffer = 2 periods
ALSA: final selected sample format for capture: 32bit integer little-endian
ALSA: use 2 periods for capture
ALSA: final selected sample format for playback: 32bit integer little-endian
ALSA: use 2 periods for playback
JackTemporaryException : now quits...
Jack main caught signal 2
OSC=osc.udp://jamiedrinkell-Nitro-AN515-57:16988/
Released audio card Audio1
audio_reservation_finish
[non-mixer-xt] Info: created by = The Non-Mixer-XT 2.0.6
[non-mixer-xt] Info: created on = Wed Dec 11 03:12:39 2024
[non-mixer-xt] Info: version = 1
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jackdmp 1.9.21
Copyright 2001-2005 Paul Davis and others.
Copyright 2004-2016 Grame.
Copyright 2016-2022 Filipe Coelho.
jackdmp comes with ABSOLUTELY NO WARRANTY
This is free software, and you are welcome to redistribute it
under certain conditions; see the file COPYING for details
no message buffer overruns
no message buffer overruns
no message buffer overruns
JACK server starting in realtime mode with priority 10
self-connect-mode is "Don't restrict self connect requests"
audio_reservation_init
Acquire audio card Audio1
creating alsa driver ... hw:PCH,0|hw:PCH,0|1024|2|48000|0|0|nomon|swmeter|-|32bit
configuring for 48000Hz, period = 1024 frames (21.3 ms), buffer = 2 periods
ALSA: final selected sample format for capture: 32bit integer little-endian
ALSA: use 2 periods for capture
ALSA: final selected sample format for playback: 32bit integer little-endian
ALSA: use 2 periods for playback
[non-mixer-xt] Add mixer strip "Guitar 1"
[non-mixer-xt] Add mixer strip "Guitar 2"
[non-mixer-xt] Add mixer strip "Bass"
[non-mixer-xt] Add mixer strip "Drums"
[non-mixer-xt] Name: Red Zeppelin Drumkit
[non-mixer-xt] Plugin has 0 AUDIO inputs and 2 AUDIO outputs
[non-mixer-xt] Plugin has 1 ATOM inputs and 1 ATOM outputs
[non-mixer-xt] Plugin has 1 MIDI in ports and 0 MIDI out ports
JackEngine::XRun: client = Non-Mixer-XT (Guitars) was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Bass was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Drums was not finished, state = Triggered
JackAudioDriver::ProcessGraphAsyncMaster: Process error
JackEngine::XRun: client = Non-Mixer-XT (Guitars) was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Bass was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Drums was not finished, state = Triggered
JackAudioDriver::ProcessGraphAsyncMaster: Process error
JackEngine::XRun: client = Non-Mixer-XT (Guitars) was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Bass was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Drums was not finished, state = Triggered
JackAudioDriver::ProcessGraphAsyncMaster: Process error
JackEngine::XRun: client = Non-Mixer-XT (Guitars) was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Bass was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Drums was not finished, state = Triggered
JackAudioDriver::ProcessGraphAsyncMaster: Process error
JackEngine::XRun: client = Non-Mixer-XT (Guitars) was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Bass was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Drums was not finished, state = Triggered
JackAudioDriver::ProcessGraphAsyncMaster: Process error
JackEngine::XRun: client = Non-Mixer-XT (Guitars) was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Bass was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Drums was not finished, state = Triggered
JackAudioDriver::ProcessGraphAsyncMaster: Process error
JackEngine::XRun: client = Non-Mixer-XT (Guitars) was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Bass was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Drums was not finished, state = Triggered
JackAudioDriver::ProcessGraphAsyncMaster: Process error
JackEngine::XRun: client = Non-Mixer-XT (Guitars) was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Bass was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Drums was not finished, state = Triggered
JackAudioDriver::ProcessGraphAsyncMaster: Process error
Cannot read socket fd = 23 err = Connection reset by peer
Could not read notification result
ClientNotify fails name = Non-Mixer-XT (Guitars) notification = 18 val1 = 0 val2 = 0
Cannot write socket fd = 26 err = Broken pipe
CheckRes error
Could not write notification
ClientNotify fails name = Non-Mixer-XT/Bass notification = 18 val1 = 0 val2 = 0
Cannot write socket fd = 29 err = Broken pipe
CheckRes error
Could not write notification
ClientNotify fails name = Non-Mixer-XT/Drums notification = 18 val1 = 0 val2 = 0
Cannot write socket fd = 29 err = Broken pipe
CheckRes error
Could not write notification
ClientNotify fails name = Non-Mixer-XT/Drums notification = 18 val1 = 1 val2 = 0
Cannot write socket fd = 26 err = Broken pipe
CheckRes error
Could not write notification
ClientNotify fails name = Non-Mixer-XT/Bass notification = 18 val1 = 1 val2 = 0
Cannot write socket fd = 23 err = Broken pipe
CheckRes error
Could not write notification
ClientNotify fails name = Non-Mixer-XT (Guitars) notification = 18 val1 = 1 val2 = 0
Cannot write socket fd = 23 err = Broken pipe
CheckRes error
Could not write notification
ClientNotify fails name = Non-Mixer-XT (Guitars) notification = 3 val1 = 0 val2 = 0
Cannot write socket fd = 26 err = Broken pipe
CheckRes error
Could not write notification
ClientNotify fails name = Non-Mixer-XT/Bass notification = 3 val1 = 0 val2 = 0
Cannot write socket fd = 29 err = Broken pipe
CheckRes error
Could not write notification
ClientNotify fails name = Non-Mixer-XT/Drums notification = 3 val1 = 0 val2 = 0
Segmentation fault (core dumped)

jamiedrinkell@jamiedrinkell-Nitro-AN515-57:~/Desktop$ JackEngine::XRun: client = Non-Mixer-XT/Bass was not finished, state = Triggered
JackEngine::XRun: client = Non-Mixer-XT/Drums was not finished, state = Triggered
Cannot write socket fd = 26 err = Broken pipe
CheckRes error
Could not write notification
ClientNotify fails name = Non-Mixer-XT (Guitars) notification = 1 val1 = 0 val2 = 0
Cannot write socket fd = 29 err = Broken pipe
CheckRes error
Could not write notification
ClientNotify fails name = Non-Mixer-XT (Guitars) notification = 1 val1 = 0 val2 = 0
JackEngine::XRun: client = Non-Mixer-XT/Drums was not finished, state = Triggered
JackAudioDriver::ProcessGraphAsyncMaster: Process error
Cannot write socket fd = 29 err = Broken pipe
CheckRes error
Could not write notification
ClientNotify fails name = Non-Mixer-XT/Bass notification = 1 val1 = 0 val2 = 0
Cannot write socket fd = 29 err = Broken pipe
CheckRes error
Could not write notification
ClientNotify fails name = Non-Mixer-XT/Drums notification = 3 val1 = 0 val2 = 0
Unknown error...
terminate called after throwing an instance of 'Jack::JackTemporaryException'
  what():

**Comment by M0JXD** [comment]
[core.non-mixer-xt.1000.1cc67d5df75f4c00bba4a233861d0d38.17439.1735315608000000.zip](https://github.com/user-attachments/files/18262021/core.non-mixer-xt.1000.1cc67d5df75f4c00bba4a233861d0d38.17439.1735315608000000.zip)

**Comment by Stazed** [comment]
The crash seems to be coming from lilv which does the state parsing. It should not be crashing if it receives an invalid directory name. I does not crash for me when when I remove an LV2 directory and correctly generates an error then ignores the state load. So I don't know why this is happening.

In addition to the missing directory for Red_Zeppelin, the project folder contains a directory for Calf_Limiter and file Compressor_Stereo which are not listed in the snapshot. If you removed them from the strip, they should have been removed when the project was saved. Clearly the snapshot and contents of the project directory are not consistent. Even still it should not be crashing in lilv. I have no idea what is going on here....

**Comment by Stazed** [comment]
Downloaded and installed Red_Zeppelin, and loaded your mixer project with no crash. So I cannot reproduce this...

**Comment by M0JXD** [comment]
Odd. Probably a faux pa of mine!
I did think it was odd that there was no plugin folder.

I have a feeling it's probably due to messing with the session a lot in unexpected ways while working on sequencer - it might be a RaySession problem instead. I think I closed the session when JACK was still playing and both Timeline and Mixer had to be force stopped.

As before with the waveform thing if it happens again I'll let you know.
